### PR TITLE
Improvements to documentation on Stream-Actor interop

### PR DESCRIPTION
**Pull request by dodnert**
There are 2 changes in this branch.

1. I have updated the text describing `Source.actorRef` to match the current, supported method with that name. The existing text describes the behavior of the deprecated two-argument method with the same name.
2. I have made a change to the Scala code example that relates to `Source.actorRefWithBackpressure`. The existing code has  an `Actor` that expects a parameter (`ackWith`) that is never used in the implementation. I don't think it is needed and its presence confused me when I studied this example.

### CODE REVIEW

**Review by raboof — APPROVED** [review]
Thanks a lot, great improvement!

**Comment by dodnert** [comment]
@raboof - Can you mark this PR as "OK TO TEST"?

**Comment by raboof** [comment]
OK TO TEST

**Comment by dodnert** [comment]
@raboof - This PR needs a second approval before it can be merged. Any suggestions? Thanks.

**Review by patriknw — APPROVED** [review]
LGTM, thanks

**Pull request merged**
